pandas/tests/plotting/test_xlim_ylim_dtype.py

```diff
@@ -0,0 +1,122 @@
+import pytest
+import pandas as pd
+from datetime import datetime as dt
+
+"""targeting #GH40781"""
+
+
+def test_if_plotable_xlim_ylim_both_ints() -> bool:
+    # checks if ValueError is raised with invalid dtype
+    df = pd.DataFrame({
+        "A": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+        "B" : [0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0]
+    })
+    try:
+        df.plot(kind="line", x="A", xlim=[1, 2], ylim=[3, 4])
+        return True
+    except ValueError:
```

Comment:
never use try/except in tests like this see simliar tests on how to check for raising

Reply:
I am looking into more pytesting, I have never written tests before so I would appreciate the guidance.

Reply:
Hi, you have done a great job in writing those tests, great work!

I think what @jreback meant is to have something like:
```python
msg = "error messages"
with pytest.raises(ValueError, match=msg):
    your_code_which_generates_error
```

There should be many references in the codebase, for instance, `https://github.com/pandas-dev/pandas/blob/master/pandas/tests/plotting/frame/test_frame.py` 

I hope this helps and please let me know if you have other questions

Reply:
Thank you, I have started working on this, I will push new changes soon.